kubernetes/resource_kubernetes_runtime_class_v1_test.go

```diff
@@ -0,0 +1,168 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesruntime_class_v1_basic(rcName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesruntime_class_v1Exists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.name", rcName),
+					resource.TestCheckResourceAttr(resourceName, "handler", "myclass"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"metadata.0.resource_version"},
+			},
+			{
+				Config: testAccKubernetesruntime_class_v1_addAnnotations(rcName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesruntime_class_v1Exists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.TestAnnotationOne", "one"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.TestAnnotationTwo", "two"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.labels.%", "0"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.name", rcName),
+					resource.TestCheckResourceAttr(resourceName, "handler", "newclass"),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.generation"),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.resource_version"),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.uid"),
+				),
+			},
+			{
+				Config: testAccKubernetesruntime_class_v1_addLabels(rcName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesruntime_class_v1Exists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.TestAnnotationOne", "one"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.annotations.TestAnnotationTwo", "two"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.labels.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.labels.TestLabelOne", "one"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.labels.TestLabelTwo", "two"),
+					resource.TestCheckResourceAttr(resourceName, "metadata.0.name", rcName),
+					resource.TestCheckResourceAttr(resourceName, "handler", "my-class"),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.generation"),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.resource_version"),
+					resource.TestCheckResourceAttrSet(resourceName, "metadata.0.uid"),
+				),
+			},
+		},
+	})
+}
+
+func testAccKubernetesruntime_class_v1_basic(name string) string {
+	return fmt.Sprintf(`resource "kubernetes_runtime_class_v1" "test" {
+  metadata {
+    name = %q
+  }
+  handler = "myclass"
```

Comment:
Since we use the runtime class resource with pods, we should have a pod using a runtimeclass resource as part of the tests.

Reply:
You don't need to create a Pod to actually test that this resource works. A RuntimeClass can exist without any Pods using it.

Reply:
@jrhouston wouldn't we want to have a test that checks that it works as expected with a Pod? I left this suggestion since not having a proper handler value would result in the Pod not being created. We could always have a message that informs the user what values to use to prevent this from happening.

Reply:
The schema already has validation for the `handler` field so they will get an error message if the handler value is not in the proper format. 

Other than that the API server doesn't produce any error if you supply a handler that doesn't exist on the node. So we don't really have a way of informing them that they've specified a bad handler value.